pkg/gui/recent_repos_panel.go

```diff
@@ -1,24 +1,83 @@
 package gui
 
 import (
+	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
+	"sync"
 
 	"github.com/jesseduffield/generics/slices"
 	"github.com/jesseduffield/lazygit/pkg/commands"
 	"github.com/jesseduffield/lazygit/pkg/env"
+	"github.com/jesseduffield/lazygit/pkg/gui/presentation/icons"
 	"github.com/jesseduffield/lazygit/pkg/gui/style"
 	"github.com/jesseduffield/lazygit/pkg/gui/types"
 )
 
+func (gui *Gui) getCurrentBranch(path string) string {
+	readHeadFile := func(path string) (string, error) {
+		headFile, err := ioutil.ReadFile(fmt.Sprintf("%s%cHEAD", path, os.PathSeparator))
+		if err == nil {
+			content := strings.TrimSpace(string(headFile))
+			branch := strings.TrimPrefix(content, "ref: refs/heads/")
+			return branch, nil
+		}
+		return "", err
+	}
+
+	gitDirPath := fmt.Sprintf("%s%c.git", path, os.PathSeparator)
+
+	if gitDir, err := os.Stat(gitDirPath); err == nil {
+		if gitDir.IsDir() {
+			// ordinary repo
+			if branch, err := readHeadFile(gitDirPath); err == nil {
+				return branch
+			}
+		} else {
+			// worktree
+			if worktreeGitDir, err := ioutil.ReadFile(gitDirPath); err == nil {
+				content := strings.TrimSpace(string(worktreeGitDir))
+				worktreePath := strings.TrimPrefix(content, "gitdir: ")
+				if branch, err := readHeadFile(worktreePath); err == nil {
+					return branch
+				}
+			}
+		}
+	}
+
+	return "HEAD"
```

Comment:
What does it actually mean if we reach here? Does it meant that there is no branch and the repo is also not on a detached head? It might be better to make it clear e.g. 'not on branch' or something

Reply:
This means that:
- there were either errors along the way (shouldn't happen, the paths are verified `git` repos for sure, unless reading the file fails) 
- or that repo is in `detached HEAD` state

I could put the short `SHA` here, actually.

Reply:
yeah that works for me

Reply:
My bad, it appears that currently the entire `SHA` was printed instead of `HEAD`.

So, `HEAD` will appear _only_ if there's an error with reading the file.

But I presume we still don't need the entire SHA, right?

Reply:
If we know that it is indeed a SHA then yeah we can truncate it. There's a helper method for that that we use when rendering commits.

If there's an error reading a file we're better off saying 'unknown branch'

Reply:
> There's a helper method for that that we use when rendering commits.

I think `fileContent[:8]` should suffice? It just contains the hash.

> If there's an error reading a file we're better off saying 'unknown branch'

Good point.

Reply:
> I18n needed?

Just this tidbit remains, but I presume you're gonna say "Yes, of course!" :)

Reply:
ah yes, of course! i18n is indeed needed